### PR TITLE
Add an `<infection>` class

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR adds a `<daedalus_infection>` class similar to the `<country_class>`. Users can still pass a pathogen/infection/epidemic name as a string if they want to use default parameters, but any modifications have to go via the class which does the input checking etc.

Infection parameters are prepared for the ODE parameter list similar to country parameters.

The class helper function `daedalus_infection()` accepts a name and `...` instead of a named list of NULL parameters because it was less typing that way; the documentation lists the accepted values, and passing an invalid parameter name will inform users about which ones are accepted.

I've removed the more complicated assignment/un-classing on invalidation which was copied from the `<daedalus_country>` class.

### CODE REVIEW

File: .lintr [hunk 1]
```diff
@@ -18,6 +18,11 @@ exclusions: list(
       object_name_linter = Inf, # due to S3 method
       object_length_linter = Inf # due to length of method name
     ),
+    "R/class_infection.R" = list(
+      duplicate_argument_linter = Inf, # due to cli_bullets
```
Comment (richfitz) [inline .lintr:22]: perhaps just remove this linter?

File: tests/testthat/test-parameters.R [hunk 2]
```diff
@@ -1,87 +1,87 @@
...
-    regexp = "Expected.*`infect_params_manual`.*single positive.*finite number"
-  )
-  expect_error(
-    daedalus(
-      country_canada, "influenza_1918",
-      infect_params_manual = list(r0 = Inf)
-    ),
-    regexp = "Expected.*`infect_params_manual`.*single positive.*finite number"
-  )
+# country_canada <- country("Canada")
```
Comment (richfitz) [inline tests/testthat/test-parameters.R:3]: Why is this commented out? Uncomment or delete?

**Review by richfitz — CHANGES_REQUESTED** [review]

File: tests/testthat/test-parameters.R (hunk 2 above)
Comment (pratikunterwegs) [reply tests/testthat/test-parameters.R:3]: Thanks for spotting - the only thing I can think of is a bad rebase? I'm surprised the linter didn't catch this as the workflow has passed. Will look at the workflow as well.

Comment (pratikunterwegs) [reply tests/testthat/test-parameters.R]: Removed in https://github.com/jameel-institute/daedalus/pull/20/commits/23e9d675f542e8bb7c18b8472fad6a4eb8b825cb

File: .lintr (hunk 1 above)
Comment (pratikunterwegs) [reply .lintr]: Moved to main exclusion list in https://github.com/jameel-institute/daedalus/pull/20/commits/198962683540135bbaa314c9fdfc8297751e41d5

**Comment by pratikunterwegs** [comment]
Thanks @richfitz - I've amended `.lintr` to make it more granular as adding any exclusion rule for `"tests"` seemed to turn off all linting in the directory. Some other issues that were caught have been fixed.

**Comment by pratikunterwegs** [comment]
Fixed the `sapply()`-like behaviour for `get_data.class()` for both country and infection class here. It still returns, say, a matrix or a vector depending on what is requested, but hopefully users are aware of that.

**Review by richfitz — APPROVED** [review]

**Pull request merged**